systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -308,13 +317,164 @@ void testKafkaNodePoolBrokerIdsManagementUsingAnnotations(ExtensionContext exten
+            k.getSpec().getKafka().getConfig().put("compression.type", "gzip");
+        }, testStorage.getNamespaceName());
+
+        StrimziPodSetUtils.waitForPrevailedPodAnnotationKeyValuePairs(testStorage.getNamespaceName(), Constants.NODE_BROKER_CONFIG_HASH_ANNOTATION, knpCConfigHashAnnotationValue, knpCStrimziPodSetName, podCName);
+
+        // replacing config relevant only to controllers
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), k -> {
+            k.getSpec().getKafka().getConfig().put("max.connections", "456");
+        }, testStorage.getNamespaceName());
+
+        StrimziPodSetUtils.waitUntilPodAnnotationChange(testStorage.getNamespaceName(), podCName, knpCStrimziPodSetName, Constants.NODE_BROKER_CONFIG_HASH_ANNOTATION, knpCConfigHashAnnotationValue);
+    }
+
+    /**
+     * Verifies that the specified configurations are correctly propagated to the Kafka Node Pool (KNP) pods.
+     *
+     * <p>This method checks the memory limit and JVM heap options of the first pod belonging to the given node pool.
+     *
+     * @param testStorage        the storage object containing test-related information.
+     * @param nodePoolName       the name of the KNP to verify.
+     * @param expectedJvmXmxLimit the expected JVM '-Xmx' heap size limit.
+     * @param expectedMemoryLimit the expected memory limit ('resource.limits.memory') for the pod.
+     *
+     * @throws AssertionError if configurations obtained from Pods do not match the expected values.
+     */
+    public static void verifyResourcesConfigurationPropagatedToKnpPods(TestStorage testStorage, String nodePoolName, String expectedJvmXmxLimit, String expectedMemoryLimit) {
+        Pod nodePoolPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(testStorage.getNamespaceName(), testStorage.getClusterName() + "-" + nodePoolName).get(0);
+        Container container = nodePoolPod.getSpec().getContainers().stream().findFirst().get();
+        Quantity  obtainedMemory = container.getResources().getLimits().get("memory");
+        EnvVar obtainedJvmHeapEnvVariable = container.getEnv().stream().filter(e -> e.getName().equals("KAFKA_HEAP_OPTS")).findAny().get();
+
+        assertThat(obtainedMemory, is(new Quantity(expectedMemoryLimit)));
+        assertThat("NodePool: does not contain expected JVM configuration", obtainedJvmHeapEnvVariable.getValue().contains(expectedJvmXmxLimit));
+    }
+
     @BeforeAll
     void setup(ExtensionContext extensionContext) {
         assumeFalse(Environment.isOlmInstall() || Environment.isHelmInstall());
 
         List<EnvVar> coEnvVars = new ArrayList<>();
         coEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+KafkaNodePools", null));
 
+        if (Environment.isKRaftModeEnabled()) {
+            coEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+UseKRaft,+KafkaNodePools", null));
+        }
+
```

Comment:
I would maybe add the `+UseKRaft` to the default and run all the tests in KRaft mode -> it will be anyway required or?